auditbeat/module/file_integrity/metricset_test.go

```diff
@@ -105,6 +105,68 @@ func TestDetectDeletedFiles(t *testing.T) {
 	}
 }
 
+func TestExcludedFiles(t *testing.T) {
+	defer setup(t)()
+
+	bucket, err := datastore.OpenBucket(bucketName)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer bucket.Close()
+
+	dir, err := ioutil.TempDir("", "audit-file")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	dir, err = filepath.EvalSymlinks(dir)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ms := mbtest.NewPushMetricSetV2(t, getConfig(dir))
+	capture := &mbtest.CapturingReporterV2{DoneC: make(chan struct{})}
+
+	go func() {
+		for _, f := range []string{"FILE.TXT", "FILE.TXT.SWP", "file.txt.swo", ".git/HEAD", ".gitignore"} {
+			file := filepath.Join(dir, f)
+			ioutil.WriteFile(file, []byte("hello world"), 0600)
+		}
+		for i := 0; i < 100 && len(capture.Events) < 3; i++ {
```

Comment:
Waiting for a certain number of events is a pretty common task in tests. We should make this easier to do with the `mb/testing` package.